decorators/autoHide.js

```diff
@@ -0,0 +1,48 @@
+var React = require('react');
+
+function autoHide(ComposedComponent) {
+  class AutoHide extends React.Component {
+    componentDidMount() {
+      this._hideOrShowContainer(this.props);
+    }
+
+    componentWillReceiveProps(nextProps) {
+      this._hideOrShowContainer(nextProps);
+    }
+
+    _hideOrShowContainer(props) {
+      var container = React.findDOMNode(this).parentNode;
+      if (props.hideWhenNoResults === true && props.hasResults === false) {
+        container.style.display = 'none';
+      } else if (props.hideWhenNoResults === true) {
+        container.style.display = '';
+      }
+    }
+
+    render() {
+      if (this.props.hasResults === false &&
+        this.props.emptyWhenNoResults === true) {
+        return <div/>;
+      }
+
+      return <ComposedComponent {...this.props} />;
+    }
+  }
+
+  AutoHide.propTypes = {
+    hasResults: React.PropTypes.bool.isRequired,
+    hideWhenNoResults: React.PropTypes.bool.isRequired,
+    emptyWhenNoResults: React.PropTypes.bool
+  };
+
+  AutoHide.defaultProps = {
+    emptyWhenNoResults: true
+  };
+
+  // precise displayName for ease of debugging (react dev tool, react warnings)
+  AutoHide.displayName = ComposedComponent.name + '-AutoHide';
```

Comment:
> Won't we end up with a displayName of stats-AutoHide-HideOnStartup-NewBehavior

Yes you will, is that an issue you think? Because here you can know in react dev tools what are the applied decorators